config/templates/metasploit-framework-wrappers/msfupdate.erb

```diff
@@ -58,23 +58,38 @@ EOF
 }
 
 install_deb() {
-	LIST_FILE=/etc/apt/sources.list.d/metasploit-framework.list
-	if [ ! -f $LIST_FILE ]; then
-		echo -n "Adding metasploit-framework to your repository list.."
-		echo "deb $DOWNLOAD_URI/apt lucid main" > $LIST_FILE
-		print_pgp_key | apt-key add -
-	fi
-	echo -n "Updating package cache.."
-	apt-get update > /dev/null
-	echo "OK"
-	echo "Checking for and installing update.."
-	apt-get install metasploit-framework
+  LIST_FILE=/etc/apt/sources.list.d/metasploit-framework.list
+  if [ ! -f $LIST_FILE ]; then
+    echo -n "Adding metasploit-framework to your repository list.."
+    echo "deb $DOWNLOAD_URI/apt lucid main" > $LIST_FILE
+    print_pgp_key | apt-key add -
+  fi
+  echo -n "Updating package cache.."
+  apt-get update > /dev/null
+  echo "OK"
+  echo "Checking for and installing update.."
+  apt-get install metasploit-framework
 
 }
 
 install_rpm() {
-	echo "Checking for and installing update.."
-	yum install metasploit-framework
+  echo "Checking for and installing update.."
+  REPO_FILE=/etc/yum.repos.d/metasploit-framework.repo
+  GPG_KEY_FILE=/etc/pki/rpm-gpg/RPM-GPG-KEY-Metasploit
+  if [ ! -f $REPO_FILE ]; then
+    echo -n "Adding metasploit-framework to your repository list.."
+
+    sh -c "cat - > /etc/yum.repos.d/metasploit-framework.repo <<EOF
```

Comment:
indeed, works! I'm not sure where that - came from, probably working around an ancient bug. Any rate, I was too slow to fix :+1: